attribute_selector_block.py

```diff
@@ -0,0 +1,84 @@
+from enum import Enum
+
+from nio.signal.base import Signal
+from nio.block.base import Block
+from nio.properties import (VersionProperty, SelectProperty, ListProperty,
+                            Property, PropertyHolder)
+
+
+class Behavior(Enum):
+    BLACKLIST = False
+    WHITELIST = True
+
+
+class SpecItem(PropertyHolder):
+    item = Property(title='Attribute')
+
+
+class AttributeSelector(Block):
+    """
+    A block for whitelisting or blacklisting incoming signals and notifying
+    the rest.
+
+    Properties:
+    specify_behavior(select): select either whitelist or blacklist behavior
+    specify_attributes(list): list of incoming signal attributes to blacklist
+                              or whitelist
+    """
+
+    version = VersionProperty('1.0.0')
+    specify_behavior = SelectProperty(Behavior, title='Specify behavior',
+                                      default=Behavior.BLACKLIST)
+    specify_attributes = ListProperty(SpecItem,
+                                      title='Incoming signal attributes',
+                                      default=[])
+
+    def __init__(self):
+        self._specify_items = None
+        super().__init__()
+
+    def start(self):
+        self._specify_items = set(spec.item() for spec in
+                                  self.specify_attributes())
+        super().start()
+
+    def process_signals(self, signals):
+        self.logger.debug('specifying these attributes: {}'
+                          .format(self._specify_items))
+
+        for index, signal in enumerate(signals):
```

Comment:
The approach of enumerating and then replacing signals at an index feels a little weird to me and wouldn't work well if our signals were passed as a general iterable. We tend to always pass lists of signals but try to keep our blocks future proof by relying on standard iteration approaches.

Normally, what we would do is start off a new empty list of signals and then add the signals we want to notify to that rather than updating the list in place.